File: src/Domain/Incomes/Income.cs
namespace Domain.Incomes;

public sealed class Income
{
    public Guid Id { get; set; }
    public Guid UserId { get; set; }
    public decimal Amount { get; set; }
    public string Source { get; set; } = string.Empty;
    public DateOnly Date { get; set; }
    public DateTime CreatedAtUtc { get; set; }

    public List<IncomeAllocation> Allocations { get; } = [];
}

File: src/Web.Client/lib/types/income.ts
export type AllocationLine = {
  accountId: string;
  amount: number;
};

export type IncomeRequest = {
  amount: number;
  source: string;
  date: string;
  allocations: AllocationLine[];
};

export type Income = {
  id: string;
  amount: number;
  source: string;
  date: string;
  allocations: AllocationLine[];
};
